src/Signal-Rohde&Schwarz_SMB100A/main.py

```diff
@@ -0,0 +1,329 @@
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# SweepMe! driver
+# * Module: Signal
+# * Instrument: Rohde&Schwarz SMB100A
+
+import numpy as np
+from pysweepme.EmptyDeviceClass import EmptyDevice
+
+
+class Device(EmptyDevice):
+    """Device class for the Rohde&Schwarz SMB100A Signal Generator."""
+
+    def __init__(self) -> None:
+        """Initialize device parameters."""
+        EmptyDevice.__init__(self)
+
+        self.port_manager = True
+        self.port_types = [
+            "USB",
+            # "GPIB",
+            "TCPIP",
+        ]
+
+        self.variables = ["Frequency", "Amplitude"]
+        self.units = ["Hz", "V"]
+        self.plottype = [True, True]
+        self.savetype = [True, True]
+
+        # Measurement Parameters
+        self.sweep_mode: str = "None"
+        self.frequency_input: str = "Frequency in Hz"  # or "Period in s"
+        self.frequency: float = 2.2e9
+
+        self.amplitude_mode: str = "Amplitude in V"  # or "High level in V"
+        self.amplitude: float = 0.1
+
+        self.delay_mode: str = "Phase in deg"  # or "Delay in s"
+        self.delay: float = 0.0
+
+    def set_GUIparameter(self) -> dict:  # noqa: N802
+        """Set the standard GUI parameters."""
+        return {
+            "SweepMode": [
+                "None",
+                "Frequency in Hz",
+                "Period in s",
+                "Amplitude in V",
+                "High level in V",
+                "Phase in deg",
+                "Delay in s",
+            ],
+            "Waveform": ["Sinus"],
+            "PeriodFrequency": ["Frequency in Hz", "Period in s"],
+            "PeriodFrequencyValue": 2e-6,
+            "AmplitudeHiLevel": ["Amplitude in V", "High level in V"],
+            "AmplitudeHiLevelValue": 1.0,
+            "DelayPhase": ["Delay in s", "Phase in deg"],
+            "DelayPhaseValue": 0.0,
+            "Impedance": ["50"],
+        }
+
+    def get_GUIparameter(self, parameter: dict) -> None:  # noqa: N802
+        """Get the GUI parameters."""
+        self.sweep_mode = parameter["SweepMode"]
+        self.frequency_input = parameter["PeriodFrequency"]
+        self.frequency = float(parameter["PeriodFrequencyValue"])
+
+        self.amplitude_mode = parameter["AmplitudeHiLevel"]
+        self.amplitude = float(parameter["AmplitudeHiLevelValue"])
```

Comment:
Because it is not clear yet whether it is an amplitude, the variable could be also called self.amplitude_highlevel_value. 
The mode could be called "self.amplitude_highlevel_mode".